packages/core/src/context.ts

```diff
@@ -3,6 +3,9 @@ import { date, number } from "./formats"
 import { isString, isFunction } from "./essentials"
 
 export const UNICODE_REGEX = /\\u[a-fA-F0-9]{4}|\\x[a-fA-F0-9]{2}/g;
+const OCT_REGEX = /(.)(#)/g;
```

Comment:
Maybe `PLURAL_PLACEHOLDER_REGEX`?
This doesn't handle the case where hash is the first character (I've slacked you a solution that works in that case).